tests/test_yaml_search_space/test_search_space.py

```diff
@@ -8,105 +8,64 @@
+        const2 = ConstantParameter(1e3, True)
+        assert const2.__eq__(pipeline_space["param_const2"]) is True
+
+    test_correct_yaml_file(BASE_PATH + "correct_config.yaml")
+    test_correct_yaml_file(BASE_PATH + "correct_config_including_types.yaml")
 
 
 @pytest.mark.neps_api
 def test_correct_including_priors_yaml_file():
     """Test the function with a correctly formatted YAML file."""
     pipeline_space = pipeline_space_from_yaml(
-        "tests/test_yaml_search_space/correct_config_including_priors.yml"
+        BASE_PATH + "correct_config_including_priors.yml"
     )
     assert isinstance(pipeline_space, dict)
-    assert isinstance(pipeline_space["learning_rate"], FloatParameter)
-    assert pipeline_space["learning_rate"].lower == 0.00001
-    assert pipeline_space["learning_rate"].upper == 0.1
-    assert pipeline_space["learning_rate"].log is True
-    assert pipeline_space["learning_rate"].is_fidelity is False
-    assert pipeline_space["learning_rate"].default == 3.3e-2
-    assert pipeline_space["learning_rate"].default_confidence_score == 0.125
-    assert isinstance(pipeline_space["num_epochs"], IntegerParameter)
-    assert pipeline_space["num_epochs"].lower == 3
-    assert pipeline_space["num_epochs"].upper == 30
-    assert pipeline_space["num_epochs"].log is False
-    assert pipeline_space["num_epochs"].is_fidelity is True
-    assert pipeline_space["num_epochs"].default == 10
-    assert pipeline_space["num_epochs"].default_confidence_score == 0.5
-    assert isinstance(pipeline_space["optimizer"], CategoricalParameter)
-    assert pipeline_space["optimizer"].choices == ["adam", 90e-3, "rmsprop"]
-    assert pipeline_space["optimizer"].is_fidelity is False
-    assert pipeline_space["optimizer"].default == 90e-3
-    assert pipeline_space["optimizer"].default_confidence_score == 4
-    assert isinstance(pipeline_space["dropout_rate"], ConstantParameter)
-    assert pipeline_space["dropout_rate"].value == 1e3
-    assert pipeline_space["dropout_rate"].default == 1e3
+    float1 = FloatParameter(0.00001, 0.1, True, False, 3.3e-2, "high")
+    assert float1.__eq__(pipeline_space["learning_rate"]) is True
+    int1 = IntegerParameter(3, 30, False, True, 10)
+    assert int1.__eq__(pipeline_space["num_epochs"]) is True
+    cat1 = CategoricalParameter(["adam", 90e-3, "rmsprop"], False, 90e-3, "medium")
+    assert cat1.__eq__(pipeline_space["optimizer"]) is True
+    const1 = ConstantParameter(1e3, True)
+    assert const1.__eq__(pipeline_space["dropout_rate"]) is True
 
 
 @pytest.mark.neps_api
 def test_incorrect_yaml_file():
     """Test the function with an incorrectly formatted YAML file."""
     with pytest.raises(SearchSpaceFromYamlFileError) as excinfo:
-        pipeline_space_from_yaml(
-            Path("tests/test_yaml_search_space/incorrect_config.txt")
-        )
+        pipeline_space_from_yaml(Path(BASE_PATH + "incorrect_config.txt"))
     assert excinfo.value.exception_type == "ValueError"
```

Comment:
I feel like this second `assert excinfo.value.exception_type == "ValueError"` isn't needed. You've nicely given a specific error type of `SearchSpaceFromYamlFileError`. Check that this error type is also a `ValueError` is not what the test is about.